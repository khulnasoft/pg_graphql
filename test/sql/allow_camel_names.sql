begin;

    comment on schema public is '@graphql({"inflect_names": false})';

    create table "AccountHolder"(
        "someId" int primary key,
        "accountHolderId" int references "AccountHolder"("someId")
    );
    comment on table "AccountHolder" is e'@graphql({"totalCount": {"enabled": true}})';

    insert into public."AccountHolder"("someId", "accountHolderId")
    values
        (1, 1),
        (2, 2);

    -- Select
    select jsonb_pretty(
        graphql.resolve($$
            {
              accountHolderCollection(first: 1) {
                pageInfo{
                  startCursor
                  endCursor
                  hasNextPage
                  hasPreviousPage
                }
                edges {
                  cursor
                  node {
                    someId
                    accountHolderId
                    accountHolder {
                      someId
                    }
                    accountHolderCollection {
                      totalCount
                    }
                  }
                }
              }
            }
        $$)
    );


    -- Paginate
    select jsonb_pretty(
        graphql.resolve($$
            {
              accountHolderCollection(first: 1 after: "WzFd" ) {
                edges {
                  cursor
                  node {
                    someId
                    accountHolderId
                    accountHolder {
                      someId
                    }
                    accountHolderCollection {
                      totalCount
                    }
                  }
                }
              }
            }
        $$)
    );

    -- Insert
    select graphql.resolve($$
    mutation {
      insertIntoAccountHolderCollection(objects: [{
        someId: 3
        accountHolderId: 2
      }]) {
        records {
          someId
          accountHolder {
            someId
          }
        }
      }
    }
    $$);

    -- Update
    select graphql.resolve($$
    mutation {
      updateAccountHolderCollection(
        set: {accountHolderId: 3}
        filter: {someId: {eq: 3}}
      ) {
        affectedCount
        records {
          someId
          accountHolder {
            someId
          }
        }
      }
    }
    $$);

    -- Delete
    select graphql.resolve($$
    mutation {
      deleteFromAccountHolderCollection(
        filter: {someId: {eq: 3}}
      ) {
        affectedCount
        records {
          someId
          accountHolder {
            someId
          }
        }
      }
    }
    $$);

rollback;
