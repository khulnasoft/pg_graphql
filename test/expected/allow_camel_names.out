begin;
    comment on schema public is '@graphql({"inflect_names": false})';
    create table "AccountHolder"(
        "someId" int primary key,
        "accountHolderId" int references "AccountHolder"("someId")
    );
    comment on table "AccountHolder" is e'@graphql({"totalCount": {"enabled": true}})';
    insert into public."AccountHolder"("someId", "accountHolderId")
    values
        (1, 1),
        (2, 2);
    -- Select
    select jsonb_pretty(
        graphql.resolve($$
            {
              accountHolderCollection(first: 1) {
                pageInfo{
                  startCursor
                  endCursor
                  hasNextPage
                  hasPreviousPage
                }
                edges {
                  cursor
                  node {
                    someId
                    accountHolderId
                    accountHolder {
                      someId
                    }
                    accountHolderCollection {
                      totalCount
                    }
                  }
                }
              }
            }
        $$)
    );
                     jsonb_pretty                     
------------------------------------------------------
 {                                                   +
     "data": {                                       +
         "accountHolderCollection": {                +
             "edges": [                              +
                 {                                   +
                     "node": {                       +
                         "someId": 1,                +
                         "accountHolder": {          +
                             "someId": 1             +
                         },                          +
                         "accountHolderId": 1,       +
                         "accountHolderCollection": {+
                             "totalCount": 1         +
                         }                           +
                     },                              +
                     "cursor": "WzFd"                +
                 }                                   +
             ],                                      +
             "pageInfo": {                           +
                 "endCursor": "WzFd",                +
                 "hasNextPage": true,                +
                 "startCursor": "WzFd",              +
                 "hasPreviousPage": false            +
             }                                       +
         }                                           +
     }                                               +
 }
(1 row)

    -- Paginate
    select jsonb_pretty(
        graphql.resolve($$
            {
              accountHolderCollection(first: 1 after: "WzFd" ) {
                edges {
                  cursor
                  node {
                    someId
                    accountHolderId
                    accountHolder {
                      someId
                    }
                    accountHolderCollection {
                      totalCount
                    }
                  }
                }
              }
            }
        $$)
    );
                     jsonb_pretty                     
------------------------------------------------------
 {                                                   +
     "data": {                                       +
         "accountHolderCollection": {                +
             "edges": [                              +
                 {                                   +
                     "node": {                       +
                         "someId": 2,                +
                         "accountHolder": {          +
                             "someId": 2             +
                         },                          +
                         "accountHolderId": 2,       +
                         "accountHolderCollection": {+
                             "totalCount": 1         +
                         }                           +
                     },                              +
                     "cursor": "WzJd"                +
                 }                                   +
             ]                                       +
         }                                           +
     }                                               +
 }
(1 row)

    -- Insert
    select graphql.resolve($$
    mutation {
      insertIntoAccountHolderCollection(objects: [{
        someId: 3
        accountHolderId: 2
      }]) {
        records {
          someId
          accountHolder {
            someId
          }
        }
      }
    }
    $$);
                                                    resolve                                                    
---------------------------------------------------------------------------------------------------------------
 {"data": {"insertIntoAccountHolderCollection": {"records": [{"someId": 3, "accountHolder": {"someId": 2}}]}}}
(1 row)

    -- Update
    select graphql.resolve($$
    mutation {
      updateAccountHolderCollection(
        set: {accountHolderId: 3}
        filter: {someId: {eq: 3}}
      ) {
        affectedCount
        records {
          someId
          accountHolder {
            someId
          }
        }
      }
    }
    $$);
                                                            resolve                                                            
-------------------------------------------------------------------------------------------------------------------------------
 {"data": {"updateAccountHolderCollection": {"records": [{"someId": 3, "accountHolder": {"someId": 3}}], "affectedCount": 1}}}
(1 row)

    -- Delete
    select graphql.resolve($$
    mutation {
      deleteFromAccountHolderCollection(
        filter: {someId: {eq: 3}}
      ) {
        affectedCount
        records {
          someId
          accountHolder {
            someId
          }
        }
      }
    }
    $$);
                                                              resolve                                                              
-----------------------------------------------------------------------------------------------------------------------------------
 {"data": {"deleteFromAccountHolderCollection": {"records": [{"someId": 3, "accountHolder": {"someId": 3}}], "affectedCount": 1}}}
(1 row)

rollback;
